spark: build packages for supported JDK versions

Dentrax opened the pull request:
Allows us to bundle the correct JDK version with the package and run tests for each supported JDK version at the package level for Spark.

Dentrax committed:
spark: build packages for supported JDK versions

Signed-off-by: Dentrax <furkan.turkal@chainguard.dev>

Dentrax commented on spark-3.5.yaml:
```diff
@@ -63,18 +65,101 @@ pipeline:
+          mkdir -p ${{targets.contextdir}}/usr/lib/spark
+          mkdir -p ${{targets.contextdir}}/usr/lib/spark/work-dir
+          mv dist/* ${{targets.contextdir}}/usr/lib/spark/
+    test:
```
Hmmm actually I'm including the testing deps in the `environment.contents.packages` using `${{range.value}}`. 🤔

Dentrax replied:
I'm going to resolve this conversation as tests are now passing, feel free to re-open if you have further thoughts on this!

Dentrax committed:
use symlink instead of cp for bitnami

Signed-off-by: Dentrax <furkan.turkal@chainguard.dev>

EyeCantCU commented on spark-3.5.yaml:
```diff
@@ -63,18 +62,105 @@ pipeline:
+    items:
+      8: "java-1.8-openjdk"
+      11: "java-11-openjdk"
+      17: "java-17-openjdk"
```
Does Spark run with OpenJDK 21? If so, we should add it here

Dentrax replied:
I don't think they "official" have support on `21`:

https://spark.apache.org/docs/latest/
> Spark runs on Java 8/11/17

And also no Java 21 on their [versions.json](https://github.com/apache/spark-docker/blob/master/versions.json).

Dentrax committed:
spark: build for java 21 and copy files for bitnami compat

Signed-off-by: Dentrax <furkan.turkal@chainguard.dev>

EyeCantCU committed:
chore(spark-3.5): Correct path to melange output dir, drop JDK 21

https://community.cloudera.com/t5/Community-Articles/Spark-and-Java-versions-Supportability-Matrix

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

Dentrax committed:
spark: move runtime deps to openjdk subpkg

Signed-off-by: Dentrax <furkan.turkal@chainguard.dev>
